### PR TITLE
relatively minor docs changes

### PR DESCRIPTION
**Pull request by herbertgithubaccount**
Currently, most of the changes are to the documentation for `table` and `string` (with a handful of exceptions).

These changes mainly consist of things I would consider to be "relatively" uncontroversial: fixing typos, adding more detail, adding `code formatting`, etc. I would be happy to change/alter things as needed.

Some of the documentation underwent a more substantive rewrite, and may have changes that are deemed more controversial. The main examples being `string.multifind`, `string.insert`, and `table.bin(search|insert)`.

Here are some of the "more noteworthy" minor changes:
* The `string.format` documentation: it now indicates that it accepts variadic arguments.
* some of the `string` methods indicate that they don't perform pattern matching.
* `tes3boundingBox` comes with a warning that bounding boxes are always centered at the origin.
* The note in `tes3worldController.musicSituation` is now formatted as an admonition tip, for consistency with other constant tables.

the documentation was rebuilt before submitting this request.

### CODE REVIEW

File: autocomplete/definitions/global/string/multifind.lua [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
+		{ name = "patterns", type = "table", description = "An array-style `table` that contains the patterns to match." },
+		{ name = "index", type = "integer", optional = true, default = 1, description = "Start index of the `find`. (Same meaning as in `string.find`.)" },
+		{ name = "plain", type = "boolean", optional = true, default = false, description = "If `true`, then a normal search will be performed instead of a pattern search. (Same meaning as in `string.find`.)" },
+	},
+	returns = {
+		{name = "pattern", type="string", optional=true, 
+			description="If a pattern was matched, then this will be the first pattern that was matched. If no patterns matched, this will be `nil`."
+		},
+		{name = "startindex", type="integer", description="If a `pattern` was matched, this is the index of `s` where the matching `pattern` begins.", optional=true,},
+		{name = "endindex", type="integer", description="If a `pattern` was matched, this is the index of `s` where the matching `pattern` ends.", optional=true,},
```
Comment (C3pa) [inline autocomplete/definitions/global/string/multifind.lua:19]: Can you add some spaces after the opening `{`, before`}` and around `=`?

File: autocomplete/definitions/global/string/split.lua [hunk 2]
```diff
@@ -1,6 +1,6 @@
 return {
 	type = "function",
-	description = "Returns an array-style table with `str` split by `sep`. The `sep`erator is not part of the results. By default the `sep`erator is `%s`, splitting the given `str`ing by spaces.",
+	description = [[Returns an array-style table with a `string` split by a specified separator. The seperator is not part of the results. By default the `sep=="%s"`, which will result in `str` getting split by whitespace characters (e.g. spaces and tabs).]],
```
Comment (C3pa) [inline autocomplete/definitions/global/string/split.lua:3]: `sep == "%s"`

File: autocomplete/definitions/global/table/bininsert.lua [hunk 3]
```diff
@@ -1,12 +1,17 @@
 return {
 	type = "function",
-	description = [[Inserts a given value through BinaryInsert into the table sorted by [, comp].
+	description = [[Inserts a given value through BinaryInsert into an array-style `table` `t`, assuming `t` was sorted by `comp`.
 
-If 'comp' is given, then it must be a function that receives two table elements, and returns true when the first is less than the second, e.g. comp = function(a, b) return a > b end, will give a sorted table, with the biggest value on position 1. [, comp] behaves as in table.sort(table, value [, comp]) returns the index where 'value' was inserted]],
+If 'comp' is given, then it must be a function that receives two values in `t` and returns `true` when the first is less than the second.
+i.e., `comp(a,b) == true` would mean `a` is less than `b`.
+
+If you want to sort in reverse order, you may set `comp=function(a,b) return a > b end`.
```
Comment (C3pa) [inline autocomplete/definitions/global/table/bininsert.lua:8]: `comp = function(a, b) return a > b end`

File: autocomplete/definitions/global/table/bininsert.lua [hunk 4]
```diff
@@ -1,12 +1,17 @@
...
+
+This function will return the index where `value` was inserted into `t`.]],
 	arguments = {
 		{ name = "t", type = "table" },
 		{ name = "value", type = "unknown" },
-		{ name = "comp", type = "unknown", optional = true },
+		{ name = "comp", type = "fun(a,b):boolean", optional = true },
```
Comment (C3pa) [inline autocomplete/definitions/global/table/bininsert.lua:14]: "fun(a, b): boolean"

File: autocomplete/definitions/global/table/binsearch.lua [hunk 5]
```diff
@@ -1,19 +1,24 @@
...
 
-If the value is found: It returns a table holding all the mathing indices (e.g. { startindice,endindice } ) endindice may be the same as startindice if only one matching indice was found
+If the `value` is found in `t`, then a `table` providing the range of all matching indices is returned. (e.g. `{ startindice, endindice }`.) 
+If only one matching index was found, then `startindice` will be the same as `endindice`.
 
-If compval is given: then it must be a function that takes one value and returns a second value2, to be compared with the input value, e.g.: compvalue = function( value ) return value[1] end
+If `value` is not found in `t`, then `nil` is returned.
 
-If reversed is set to true: then the search assumes that the table is sorted in reverse order (largest value at position 1) note when reversed is given compval must be given as well, it can be nil/_ in this case
+If `compval` is given, then it must be a function that takes one value and returns the value to use for comparisons. For example, to compare arrays based on their first entry, you can write `compvalue = function( value ) return value[1] end`.
```
Comment (C3pa) [inline autocomplete/definitions/global/table/binsearch.lua:10]: I see that this is as it was before, but let's  now update it to be consistent with the rest of the project `compvalue = function(value) return value[1] end`

File: autocomplete/definitions/global/table/binsearch.lua [hunk 6]
```diff
@@ -1,19 +1,24 @@
...
+If `reversed == true`, then the search assumes that `t` is sorted in reverse order (i.e., largest value at position 1).
+Note that specifying `reversed` requires specifying `compval`. 
+You can circumvent this by passing `nil` for `compval`. e.g., `binsearch(tbl, value, nil, true)`.]],
 	arguments = {
 		{ name = "t", type = "table" },
-		{ name = "value", type = "unknown" },
-		{ name = "compval", type = "unknown", optional = true },
-		{ name = "reversed", type = "unknown", optional = true },
+		{ name = "value", type = "unknown", description="The value to search for." },
+		{ name = "compval", type = "function", optional = true, description="A function that returns the value to use in comparisons." },
```
Comment (C3pa) [inline autocomplete/definitions/global/table/binsearch.lua:20]: Please add spaces around `=`.

File: autocomplete/definitions/global/table/binsearch.lua [hunk 7]
```diff
@@ -1,19 +1,24 @@
...
+		{ name = "reversed", type = "boolean", optional = true, "If true, then `binsearch` will assume `t` is sorted in reverse order." },
```
Comment (C3pa) [inline autocomplete/definitions/global/table/binsearch.lua:21]: You need to add `description = `.

File: autocomplete/definitions/global/table/empty.lua [hunk 8]
```diff
@@ -1,9 +1,11 @@
 return {
 	type = "function",
-	description = [[Returns true if the table is empty, otherwise, it returns false.]],
+	description = [[Checks if a table is empty. 
+	
+	If `deepCheck==true`, then tables are allowed to have nested subtables, so long as those subtables are empty. e.g., `table.empty({ {}, {} }, true) == true`, while `table.empty({ {}, {} }) == false`.]],
```
Comment (C3pa) [inline autocomplete/definitions/global/table/empty.lua:5]: `deepCheck == true`

Comment (herbertgithubaccount) [reply autocomplete/definitions/global/table/empty.lua]: i made another commit in an attempt to address this and all the suggestions. sorry for messing up the spacing again, i made most of the changes in this PR before i saw your feedback about spacing in the variadic types PR.

**Review by C3pa — APPROVED** [review]

**Comment by C3pa** [comment]
Thanks, Herbert!

**Pull request merged**